### PR TITLE
Documentation: Add conventions doc

### PR DESCRIPTION
**Pull request by tomdee**

### CODE REVIEW

**Comment by tomdee** [comment]
@containernetworking/cni-maintainers I've updated based on our discussions on Wednesday. PTAL.

File: CONVENTIONS.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   "ipam":{  
+     <IPAM CONFIG HERE>
+   }
+}
+```
+
+| Area  | Purpose| Spec and Example | Runtime implementations | Plugin Implementations |
+| ------ | ------ | ------             | ------  | ------                  | ------                 |  
+| labels | Pass`key=value` labels to plugins | <pre>"labels" : [<br />  { "key" : "app", "value" : "myapp" },<br />  { "key" : "env", "value" : "prod" }<br />] </pre> | none | none |
+| port mappings | Pass mapping from ports on the host to ports in the container network namespace. | <pre>"port_mappings" : [<br />  { "host_port": 8080, "container_port": 80, "protocol": "tcp" },<br />  { "host_port": 8000, "container_port": 8001, "protocol": "udp" }<br />] </pre> | none | none |
```
Comment (rosenhouse) [inline CONVENTIONS.md]: "port mappings" is also listed above as a "plugin specific" field.  If I recall correctly, the consensus from our meeting this week was that we preferred it live there, and not here.  Maybe remove this row from this table?

Comment (tomdee) [inline CONVENTIONS.md]: I though I'd already removed it from this table, oops. Will push an update for that when I get a chance.

**Comment by rosenhouse** [comment]
LGTM!

**Pull request merged**